src/api/firebase.js

```diff
@@ -72,10 +73,67 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem(listId, itemId, currentItemUpdates) {
+export async function updateItem(
+	listId,
+	itemId,
+	dateLastPurchased,
+	dateNextPurchased,
+	dateCreated,
+	currentItemUpdates,
+) {
 	const currentItemRef = doc(db, listId, itemId);
+	// use updated totalPurchases
+	const { totalPurchases } = currentItemUpdates;
+
+	let daysSinceLastTransaction;
+	let previousEstimate;
+	// When the item was purchased first time
+	if (totalPurchases === 1) {
+		// number of days from the option selected by the user
+		daysSinceLastTransaction = getDaysBetweenDates(
+			dateCreated.toDate(),
+			dateNextPurchased.toDate(),
+		);
+	}
+	// after the first purchase
+	else {
+		previousEstimate = getDaysBetweenDates(
+			dateLastPurchased.toDate(),
+			dateNextPurchased.toDate(),
+		);
+		daysSinceLastTransaction = getDaysBetweenDates(
+			dateLastPurchased.toDate(),
+			new Date(),
+		);
+	}
+	const numberOfDays = calculateEstimate(
+		previousEstimate,
+		daysSinceLastTransaction,
+		totalPurchases,
+	);
+
+	const estimatedNextPurchaseDate = getFutureDate(numberOfDays);
+
+	console.log(
+		'previousEstimate: ',
+		previousEstimate,
+		'\ndaysSinceLastTransaction:',
+		daysSinceLastTransaction,
+		'\ndays to dateNextPurchased(Rounded Average of the last two values): ',
+		numberOfDays,
+		'\ntotalPurchases: ',
+		totalPurchases,
+		'\nToday: ',
+		new Date(),
+		'\nestimatedNextPurchaseDate: ',
+		estimatedNextPurchaseDate,
+	);
+
 	try {
-		await updateDoc(currentItemRef, currentItemUpdates);
+		await updateDoc(currentItemRef, {
+			...currentItemUpdates,
+			dateNextPurchased: estimatedNextPurchaseDate,
```

Comment:
**praise**: Nice work using the `...` spread operator to add a property to this object - very easy to read!